app/controllers/application_controller.rb

```diff
@@ -1,5 +1,15 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    User.first
+  # def current_user
+  #   User.first
+  # end
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:email, :password, :current_password) }
```

Comment:
- [ ] You suppose to authenticate all controllers before working with any of them, So please use 
`before_action :authenticate_user!` here in the application controller and that will fix the issue that the previous reviewer had, and will direct you to the login page when you run the rails server.